matrix_client/crypto/olm_device.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+import olm
```

Comment:
This is a bit different, here we can have the import throw ImportError. Also I think it's better to put it farther away: OlmDevice has the right to assume that an application using it has made sure that its dependencies are satisfied. Also, I end up with multiple files (4, currently) with `import olm`, and we don't want to duplicate the checking code 4 times.  
Here's my solution to this problem from a later PR https://github.com/matrix-org/matrix-python-sdk/pull/234/commits/b2e9ab3b1d29076fc296c3b9cd6eab099f51c09a. Let me know what you think!

Reply:
I absolutely share this concern, but to my understanding this PR wouldn't break master, and  `import matrix_client` doesn't fail without olm. Only `import matrix_client.crypto.olm_device` will, and the above commit will not change that, so I don't really see what the use of cherry-picking it is?

I'll move the import code.

Reply:
The other thing here is that I'd like tests to still run if olm isn't available... Maybe just guard the entire crypto test file?

Reply:
>I absolutely share this concern, but to my understanding this PR wouldn't break master, and  `import matrix_client` doesn't fail without olm. Only `import matrix_client.crypto.olm_device` will, and the above commit will not change that, so I don't really see what the use of cherry-picking it is?
>
>I'll move the import code.

You're right. I misread. I though that when I was playing with this yesterday, I got an error on `import matrix_client`, but I must have been mistaken.

Reply:
Yup, I thought about this and found out that pytest offers just the right feature https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency.  
But the I thought some more, and I wasn't sure that we really wanted to ignore tests implicitly like that. Especially since this is really easy to do: `pytest --ignore=test/crypto`.

Just tell me, you have the final word on this :)

Reply:
People should still be able to run tests if they don't have every feature of the sdk installed. Final world spoken. ;)